test/e2e/common/node/container_probe.go

```diff
@@ -756,12 +756,12 @@ func newInformerWatchPod(ctx context.Context, c clientset.Interface, podNamespac
 		&cache.ListWatch{
 			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
 				options.FieldSelector = fields.SelectorFromSet(fields.Set{"metadata.name": podName}).String()
-				obj, err := c.CoreV1().Pods(podNamespace).List(ctx, options)
+				obj, err := c.CoreV1().Pods(podNamespace).List(context.Background(), options)
```

Comment:
I suspect that this is the wrong place to fix this issue. Either `newInformerWatchPod` shouldn't take a context and then always use `context.Background()`, or the callers of `newInformerWatchPod` need to be fixed to pass in a correct context that remains valid as long as the watch is needed.

Reply:
I'll have a look tomorrow.

Reply:
but the point is that the ListWatch function context is unrelated to anything external, since they are governed by the controller.Run() loop, if that is not stopped they will keep connection/cancelled/retrying

Reply:
but let me remove this commit and discuss it tomorrow, as you say, it is only a problem if the infomer is not stopped ... but if the informer is not stopped means that we are leaking an infomrer , right ?